js/packages/cli/src/candy-machine-cli.ts

```diff
@@ -100,22 +108,47 @@ programCommand('upload')
       retainAuthority,
       mutable,
       rpcUrl,
+      arweaveJwk,
       batchSize,
     } = cmd.opts();
 
-    if (storage === 'ipfs' && (!ipfsInfuraProjectId || !ipfsInfuraSecret)) {
+    if (storage === StorageType.ArweaveSol && env !== 'mainnet') {
+      throw new Error(
+        'Arweave Sol must receive real SOL from mainnet, for devnet please use Arweave Bundle, AWS or IPFS\n',
```

Comment:
```suggestion
        'The arweave-sol storage option only works on mainnet. For devnet, please use either arweave, aws or ipfs\n',
```

Reply:
Please add a similar mainnet check for `ArweaveBundle` since that also requires spending real tokens - doing so signals a misconfiguration by the user.